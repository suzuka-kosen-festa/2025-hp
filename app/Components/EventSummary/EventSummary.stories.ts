import type { Meta, StoryObj } from "@storybook/react";
import EventSummary from "./EventSummary";

type T = typeof EventSummary;

const meta: Meta<T> = {
  title: "Components/EventSummary",
  component: EventSummary,
  parameters: {
    layout: "fullscreen",
  },
  tags: ["autodocs"],
};

export default meta;
type Story = StoryObj<T>;

export const Default: Story = {
  args: {
    mainTitle: "Orbit",
    description: "テーマの説明ウンタラカンタラ",
    dateLabel: "開催日時",
    date: "2024年11月2日(土)～11月3日(日)",
    locationLabel: "開催場所",
    location: "鈴鹿工業高等専門学校",
  },
};
