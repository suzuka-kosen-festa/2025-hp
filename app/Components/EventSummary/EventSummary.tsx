import type { ReactNode } from "react";
import { Box, Typography } from "@mui/material";

interface EventSummaryProps {
  mainTitle?: string
  description?: string
  dateLabel: string
  date: string
  locationLabel: string
  location: string
}

function EventSummary({
  mainTitle,
  description,
  dateLabel,
  date,
  locationLabel,
  location,
}: EventSummaryProps): ReactNode {
  return (
    <Box
      sx={{
        backgroundColor: "none",
        // backdropFilter: "blur(5px)",
        color: "white",
        p: { xs: 2, sm: 3, md: 4 },
        fontFamily: "'Noto Sans JP', -apple-system, BlinkMacSystemFont, 'Segoe UI', 'Roboto', 'Oxygen', 'Ubuntu', 'Cantarell', 'Fira Sans', 'Droid Sans', 'Helvetica Neue', sans-serif",
      }}
    >
      {mainTitle && (
        <Typography
          variant="h2"
          component="h1"
          sx={{
            textAlign: "center",
            mb: 2,
            fontSize: { xs: "1.5rem", sm: "2rem", md: "2.5rem" },
          }}
        >
          {mainTitle}
        </Typography>
      )}
      {description && (
        <Typography sx={{
          textAlign: "center",
          mb: { xs: 4, sm: 6, md: 8 },
          fontSize: { xs: "0.9rem", sm: "1rem" },
        }}
        >
          {description}
        </Typography>
      )}

      <Box sx={{ mb: { xs: 3, sm: 4 } }}>
        <Box sx={{ display: "inline-block", borderBottom: "1px solid orange", pb: 0.5, mb: 2 }}>
          <Typography
            variant="h5"
            component="h2"
            sx={{
              fontSize: { xs: "1.1rem", sm: "1.25rem", md: "1.5rem" },
            }}
          >
            {dateLabel}
          </Typography>
        </Box>
        <Typography
          variant="h4"
          sx={{
            textAlign: "center",
            fontSize: { xs: "1.25rem", sm: "1.5rem", md: "2rem" },
          }}
        >
          {date}
        </Typography>
      </Box>

      <Box>
        <Box sx={{ display: "inline-block", borderBottom: "1px solid orange", pb: 0.5, mb: 2 }}>
          <Typography
            variant="h5"
            component="h2"
            sx={{
              fontSize: { xs: "1.1rem", sm: "1.25rem", md: "1.5rem" },
            }}
          >
            {locationLabel}
          </Typography>
        </Box>
        <Typography
          variant="h4"
          sx={{
            textAlign: "center",
            fontSize: { xs: "1.25rem", sm: "1.5rem", md: "2rem" },
          }}
        >
          {location}
        </Typography>
      </Box>
    </Box>
  );
}

export default EventSummary;
