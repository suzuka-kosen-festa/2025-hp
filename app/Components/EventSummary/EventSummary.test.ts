// @vitest-environment jsdom

import { composeStories } from "@storybook/react";
import { expect, it } from "vitest";
import * as stories from "./EventSummary.stories";

const { Default } = composeStories(stories);

it("should render the event summary correctly", async () => {
  await Default.run();
  expect(document.body.firstChild).toMatchSnapshot();
});
