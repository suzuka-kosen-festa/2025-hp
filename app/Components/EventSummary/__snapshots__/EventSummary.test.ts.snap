// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render the event summary correctly 1`] = `
<div>
  <div
    class="MuiBox-root css-c4cims"
  >
    <h1
      class="MuiTypography-root MuiTypography-h2 css-f7v187-MuiTypography-root"
    >
      Orbit
    </h1>
    <p
      class="MuiTypography-root MuiTypography-body1 css-f5dy3o-MuiTypography-root"
    >
      テーマの説明ウンタラカンタラ
    </p>
    <div
      class="MuiBox-root css-9i8z7t"
    >
      <div
        class="MuiBox-root css-jvpn1k"
      >
        <h2
          class="MuiTypography-root MuiTypography-h5 css-1oddxoz-MuiTypography-root"
        >
          開催日時
        </h2>
      </div>
      <h4
        class="MuiTypography-root MuiTypography-h4 css-s8m2ku-MuiTypography-root"
      >
        2024年11月2日(土)～11月3日(日)
      </h4>
    </div>
    <div
      class="MuiBox-root css-0"
    >
      <div
        class="MuiBox-root css-jvpn1k"
      >
        <h2
          class="MuiTypography-root MuiTypography-h5 css-1oddxoz-MuiTypography-root"
        >
          開催場所
        </h2>
      </div>
      <h4
        class="MuiTypography-root MuiTypography-h4 css-s8m2ku-MuiTypography-root"
      >
        鈴鹿工業高等専門学校
      </h4>
    </div>
  </div>
</div>
`;
